src/measure_theory/lp_space.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2020 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Rémy Degenne.
+-/
+import measure_theory.l1_space
+import analysis.special_functions.pow
+
+/-!
+# ℒp space
+
+This file describes properties of measurable functions with finite seminorm
+`(∫⁻ a, (nnnorm (f a))^(p : ℝ) ∂μ) ^ (1/p)` for `p:ℝ` with `1 ≤ p`.
+
+## Main definitions
+
+* `mem_ℒp f p μ` : the function `f` has finite p-seminorm for measure `μ`, for `p:ℝ` such that
+                  `hp1 : 1 ≤ p`,
+
+## Notation
+
+* `snorm f p μ` : `(∫⁻ a, (nnnorm (f a))^(p : ℝ) ∂μ) ^ (1/p)` for `f : α → β`
+
+-/
+
+open measure_theory
+
+noncomputable theory
+
+namespace ℒp_space
+
+variables {α β γ: Type*} [measurable_space α] {μ : measure α}
+  [measurable_space β] [normed_group β]
+  [normed_group γ]
+  {p : ℝ}
+
+section ℒp_space_definition
+
+/-- The property that f belongs to ℒp for measure μ and real p -/
+def mem_ℒp (f : α → β) (p : ℝ) (μ : measure α) : Prop :=
+measurable f ∧ ∫⁻ a, (nnnorm (f a)) ^ p ∂μ < ⊤
+
+/-- seminorm on ℒp -/
+def snorm (f : α → γ) (p : ℝ) (μ : measure α) : ennreal := (∫⁻ a, (nnnorm (f a))^p ∂μ) ^ (1/p)
+
+lemma lintegral_rpow_nnnorm_eq_rpow_snorm {f : α → γ} (hp0_lt : 0 < p) :
+  ∫⁻ a, (nnnorm (f a)) ^ p ∂μ = (snorm f p μ) ^ p :=
+begin
+  unfold snorm,
+  rw [←ennreal.rpow_mul, one_div, inv_mul_cancel, ennreal.rpow_one],
+  symmetry, exact ne_of_lt hp0_lt,
+end
+
+end ℒp_space_definition
+
+lemma mem_ℒp_one_iff_integrable : ∀ f : α → β, mem_ℒp f 1 μ ↔ integrable f μ :=
+begin
+  intro f,
+  unfold integrable, unfold has_finite_integral, unfold mem_ℒp,
```

Comment:
We try to have only one instruction per line, so this line could be instead `unfold integrable has_finite_integral mem_ℒp,`. Even better, these unfolds can go into the `simp only` line, turning your proof into a one-liner if you have also removed the `intro f` according to my previous comment.